datadog_checks_dev/datadog_checks/dev/tooling/commands/env/start.py

```diff
@@ -68,18 +68,20 @@
 @click.option('--org-name', '-o', help='The org to use for data submission.')
 @click.option('--profile-memory', '-pm', is_flag=True, help='Whether to collect metrics about memory usage')
 @click.option('--dogstatsd', is_flag=True, help='Enable dogstatsd port on agent')
+@click.option('--disable-agent-update', is_flag=True, help='Do not try to update the agent image')
```

Comment:
how does this interact with the `--agent` option?

Reply:
Ah I see. Maybe "updating" was not the correct word in that case then. What ddev is doing is that it `docker pull` the exact image you specified with the `-a` parameter, no matter what. In most cases, we do not have the image locally so it makes sense, but in my case this image only exists locally, so I don't want to `pull` anything.

Reply:
I wonder if instead of adding a flag we can make it so that we only pull an image when it's actually necessary (i.e. when it's not available locally).

Reply:
Hahaha, that's *exactly* what @FlorentClarret  and me just agreed on in slack: